### ISSUE TITLE
1.11 and above: optimizers function incorrectly with datasets with unbalanced labels

### ISSUE THREAD

**Issue by VOvchinnikov** [tensorflow/tensorflow#29534]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.10 (16.04 has the same issue)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): `('v1.11.0-0-gc19e29306c', '1.11.0')`, `('v1.12.0-0-ga6d8ffae09', '1.12.0')`, `('v1.13.1-0-g6612da8951', '1.13.1')`
- Python version: 2.7, 3.5, 3.6, 3.7
- Bazel version (if compiling from source): -
- GCC/Compiler version (if compiling from source): -
- CUDA/cuDNN version: -
- GPU model and memory: - (reproducable with GPU explicitly disabled)

**Describe the current behavior**
- Reference version: `('v1.10.0-0-g656e7a2b34', '1.10.0')`

We are experiencing the issue on our models, that used to be trained using Tensorflow 1.10, where there was no issue, but since we upgraded to 1.13, it appeared.
I have constructed an artificial, as simple as possible example, showcasing the issue.

The example in the attached Zip file is using `imports85` data. The "True" label is `price < PRICE_CUTOFF`, with `PRICE_CUTOFF` set so that the resulting label mean is `0.0146`, making it significantly unbalanced.

The FTRL optimizer in the example code has bizarre coefficients set to demonstrate the issue more graphically. In our models we have them in reasonable range and still get the wrong behavior.

If the model is trained and evaluated using *1.10*, the results (given all the constants as set in the code) are the following:
```
Resulting means:
  Label: 0.0146341463415
  Prediction (no optimizer): 0.023864556
  Prediction (optimizer): 0.026022965
```

Now, the results above are pretty bad, but it's due to really small dataset, used to train the model, but it's not significant from the standpoint of showcasing wrong behavior: in our models, the predictions, done with 1.10-trained model are pretty on-point.

When the model is trained and evaluated using *1.13*, the model *WITH* optimizer applied starts to show unreasonably high predictions:
```
Resulting means:
  Label: 0.0146341463415
  Prediction (no optimizer): 0.02390607
  Prediction (optimizer): 0.044271074
``` 
These results above are consistent with what we see in our models, where predictions became overwhelmingly positive once 1.13 was used for the model training (since we do use optimizers).

This does not seem to be an issue with FTRL optimizer itself, since we have tried several of them, all showing significant skewing of predicted results towards one or other end, if the label in the dataset is not balanced.

The example code should allow easy reproduction of the issue. Zip file also contains two requirements files: `requirements_1.10.txt` and `requirements_1.13.txt` to streamline setting up the virtual environments to test the issue.

**Code to reproduce the issue**
See attached Zip
[optimizers_unbalanced_label_bug.zip](https://github.com/tensorflow/tensorflow/files/3266358/optimizers_unbalanced_label_bug.zip)

**Comment by VOvchinnikov** [comment]
I could test with the other TF versions to better identify, where stuff went south - and it seems to be `1.11` (I have tried `('v1.11.0-0-gc19e29306c', '1.11.0')` specifically) version. For it, the results are much like for 1.13:
```
Resulting means:
  Label: 0.0146341463415
  Prediction (no optimizer): 0.023945015
  Prediction (optimizer): 0.044300616
```
Wasn't it a version, where some new graph optimization mechanisms were introduced?

**Comment by VOvchinnikov** [comment]
...and the same in 1.12 (`('v1.12.0-0-ga6d8ffae09', '1.12.0')`):
```
Resulting means:
  Label: 0.0146341463415
  Prediction (no optimizer): 0.023909163
  Prediction (optimizer): 0.044271577
```

Apparently, something was introduced in 1.11 and broke stuff.

**Comment by ravikyram** [comment]
@VOvchinnikov I tried reproducing issue with the provided code ,but i am unable to reproduce .Will it be possible to provide minimal code snippet which can reproduce the reported issue.Thanks

**Comment by VOvchinnikov** [comment]
Closing the issue, because the observed effect has been caused by this: https://github.com/tensorflow/tensorflow/issues/18317

The deviation of prediction mean, when using other optimizers, was caused by my mistake during setup of the experiment.
